src/GitPrompt.ps1

```diff
@@ -142,42 +221,72 @@ function Write-Prompt {
         $BackgroundColor = $null,
         [parameter(ValueFromPipeline = $true)]
         [Text.StringBuilder]
-        $Builder)
+        $Builder
+    )
+
     $s = $global:GitPromptSettings
-    if ($s -and ($null -eq $ForegroundColor)) {
-        $ForegroundColor = $s.DefaultForegroundColor
+    if ($s) {
+        if ($null -eq $ForegroundColor) {
+            $ForegroundColor = $s.DefaultColor.ForegroundColor
+        }
+
+        if ($null -eq $BackgroundColor) {
+            $BackgroundColor = $s.DefaultColor.BackgroundColor
+        }
+    }
+
+    if ($Object -is [PoshGitCellColor]) {
+        Write-Warning "Unexpected PoshGitCellColor passed to Write-Prompt"
     }
 
     if ($GitPromptSettings.AnsiConsole) {
-        $e = [char]27 + "["
-        $f = Get-ForegroundVirtualTerminalSequence $ForegroundColor
-        $b = Get-BackgroundVirtualTerminalSequence $BackgroundColor
+        if ($Object -is [PoshGitTextSpan]) {
+            $str = $Object.RenderAnsi()
+        }
+        else {
+            $e = [char]27 + "["
+            $f = Get-ForegroundVirtualTerminalSequence $ForegroundColor
+            $b = Get-BackgroundVirtualTerminalSequence $BackgroundColor
+            $str = "${f}${b}${Object}${e}0m"
```

Comment:
On second thought, wrapping text in a color seems like it would be more appropriately `PoshGitCellColor`'s responsibility, assuming we teach `Write-Prompt` to receive a `PoshGitColorCell` instead of separate properties.